ibc-clients/ics08-wasm/types/src/msgs/store_code.rs

```diff
@@ -0,0 +1,38 @@
+use ibc_primitives::prelude::*;
+use ibc_primitives::Signer;
+use ibc_proto::ibc::lightclients::wasm::v1::MsgStoreCode as RawMsgStoreCode;
+use ibc_proto::Protobuf;
+
+use crate::error::Error;
+use crate::Bytes;
+
+pub const STORE_CODE_TYPE_URL: &str = "/ibc.lightclients.wasm.v1.MsgStoreCode";
+
+/// Defines the message type for storing the Wasm byte code on the chain.
+#[derive(Clone, PartialEq, Debug, Eq)]
+pub struct MsgStoreCode {
+    pub signer: Signer,
+    pub wasm_byte_code: Bytes,
+}
+
+impl Protobuf<RawMsgStoreCode> for MsgStoreCode {}
+
+impl From<MsgStoreCode> for RawMsgStoreCode {
+    fn from(value: MsgStoreCode) -> Self {
+        Self {
+            signer: value.signer.to_string(),
+            wasm_byte_code: value.wasm_byte_code,
+        }
+    }
+}
+
+impl TryFrom<RawMsgStoreCode> for MsgStoreCode {
+    type Error = Error;
+
+    fn try_from(value: RawMsgStoreCode) -> Result<Self, Self::Error> {
+        Ok(Self {
+            signer: Signer::from(value.signer),
+            wasm_byte_code: value.wasm_byte_code,
+        })
```

Comment:
Since `try_from` doesn't return any error, we can simply implement `From` which auto implements `TryFrom<Error = Infallible>`.

Reply:
This `TryFrom` implementation is imposed by the `Protobuf` bounds ([this](https://github.com/informalsystems/tendermint-rs/blob/cc04d6bdfcab8b8d6bacaa664e6b99ea8bc40890/proto/src/lib.rs#L115)). 
However, we can move away with `Error = Infallible`. 👍🏻

Reply:
[4e86e31](https://github.com/cosmos/ibc-rs/pull/1031/commits/4e86e31831436f63dc51a76e33a96178a33cdb64)